components/collector/tests/source_collectors/trivy/test_security_warnings.py

```diff
@@ -0,0 +1,85 @@
+
+
+class TrivyJSONSecurityWarningsTest(SourceCollectorTestCase):
+    """Unit tests for the security warning metric."""
+
+    SOURCE_TYPE = "trivy_json"
+    METRIC_TYPE = "security_warnings"
+    VULNERABILITIES_JSON: ClassVar[TrivyJSON] = [
+        {
+            "Target": "php-app/composer.lock",
+            "Vulnerabilities": None,
+        },
+        {
+            "Target": "trivy-ci-test (alpine 3.7.1)",
+            "Vulnerabilities": [
+                {
+                    "VulnerabilityID": "CVE-2018-16840",
+                    "PkgName": "curl",
+                    "InstalledVersion": "7.61.0-r0",
+                    "FixedVersion": "7.61.1-r1",
+                    "Title": 'curl: Use-after-free when closing "easy" handle in Curl_close()',
+                    "Description": "A heap use-after-free flaw was found in curl versions from 7.59.0 through ...",
+                    "Severity": "HIGH",
+                    "References": [
+                        "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-16840",
+                    ],
+                },
+                {
+                    "VulnerabilityID": "CVE-2019-3822",
+                    "PkgName": "curl",
+                    "InstalledVersion": "7.61.1-r0",
+                    "FixedVersion": "7.61.2-r2",
+                    "Title": "curl: NTLMv2 type-3 header stack buffer overflow",
+                    "Description": "libcurl versions from 7.36.0 to before 7.64.0 are vulnerable to ...",
+                    "Severity": "MEDIUM",
+                    "References": [
+                        "https://curl.haxx.se/docs/CVE-2019-3822.html",
+                        "https://lists.apache.org/thread.html",
+                    ],
+                },
+            ],
+        },
+    ]
+    EXPECTED_ENTITIES: ClassVar[list[dict[str, str]]] = [
+        {
+            "key": "CVE-2018-16840@curl@trivy-ci-test (alpine 3_7_1)",
+            "vulnerability_id": "CVE-2018-16840",
+            "title": 'curl: Use-after-free when closing "easy" handle in Curl_close()',
+            "description": "A heap use-after-free flaw was found in curl versions from 7.59.0 through ...",
+            "level": "HIGH",
+            "package_name": "curl",
+            "installed_version": "7.61.0-r0",
+            "fixed_version": "7.61.1-r1",
+            "url": "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-16840",
+        },
+        {
+            "key": "CVE-2019-3822@curl@trivy-ci-test (alpine 3_7_1)",
+            "vulnerability_id": "CVE-2019-3822",
+            "title": "curl: NTLMv2 type-3 header stack buffer overflow",
+            "description": "libcurl versions from 7.36.0 to before 7.64.0 are vulnerable to ...",
+            "level": "MEDIUM",
+            "package_name": "curl",
+            "installed_version": "7.61.1-r0",
+            "fixed_version": "7.61.2-r2",
+            "url": "https://curl.haxx.se/docs/CVE-2019-3822.html",
```

Comment:
Ahh, so here there are multiple urls. How do we know that the first one in the list is the important one?

Reply:
We don't. The assumption is that it's at least relevant ;-)